### DISCUSSION TITLE
Use !empty() instead of count() > 0 where possible

### DISCUSSION THREAD

**Discussion by tmotyl** [OpenMage/magento-lts#2159] [Ideas]
In many places magento calls count() just to check if array is empty.
These places should be changed to use !empty() instead.

**Comment by sreichel** [comment]
Whats the best way to check for non-/empty array?

If you want wo compare to `null`, you do not use `if ($var)` or `if (is_null($var))`, you use `if ($var === null)`. I'd prefer to use same approach for `array`.

`(bool) $arr` works, but is hard to read, same for `empty($arr)`, because you don't know (or see) the expected/assumed type. I'd prefer plain comparison to an empty array .... `$arr === array()`. Faster than `count()` and better readable than `empty()`.

**Comment by loekvangool** [comment]
Not a whole lot of difference between `empty($arr) === false` and `$arr === array()`, performance wise: https://3v4l.org/kIDbg

**Comment by colinmollenhour** [comment]
My personal preference is `if ($arr)` or `if ( ! empty($arr))` but I really don't see a problem with using `count`. For example, what if the code is:

```php
if (count($arr) === 1) {
  //
} elseif (count($arr) === 2) {
  //
} elseif (count($arr) > 0) {
  //
}
```

To me that is more readable than the proposed:

```php
if (count($arr) === 1) {
  //
} elseif (count($arr) === 2) {
  //
} elseif ( ! empty($arr)) {
  //
}
```

**Comment by loekvangool** [comment]
`count` is orders of magnitude slower than `empty`

I would also vote for using `===` wherever possible.

**Comment by sreichel** [comment]
Worth a read .... https://www.drupal.org/project/features/issues/3074109

**Comment by sreichel** [comment]
> To me that is more readable than the proposed: .... 

Just from readablity i'd also prefer `count()`, but knowing exact size is not needed in many places. If we want to change it and unify code i'd prefer `$arr === []`.

**Comment by simbus82** [comment]
> `count` is orders of magnitude slower than `empty`
> 
> I would also vote for using `===` wherever possible.

http://webcache.googleusercontent.com/search?q=cache:2XsKOj_gf8QJ:webjawns.com/2014/04/php-benchmark-count-vs-empty/+&cd=2&hl=it&ct=clnk

Exactly, "empty" is a right choice.
